profiles/fscloud/main.tf

```diff
@@ -0,0 +1,155 @@
+  hmac_key_name                 = var.hmac_key_name
+  hmac_key_role                 = var.hmac_key_role
+  cos_location                  = var.cos_location
+  cos_plan                      = var.cos_plan
+  cos_tags                      = var.cos_tags
+  sysdig_crn                    = var.sysdig_crn
+  activity_tracker_crn          = var.activity_tracker_crn
+  instance_cbr_rules            = var.instance_cbr_rules
+}
+
+# Create IAM Authorization Policies to allow COS to access kms for the encryption key
+resource "ibm_iam_authorization_policy" "primary_kms_policy" {
+  count                       = var.skip_iam_authorization_policy ? 0 : 1
+  source_service_name         = "cloud-object-storage"
+  source_resource_instance_id = module.cos_instance.cos_instance_guid
+  target_service_name         = "hs-crypto"
+  target_resource_instance_id = var.primary_existing_hpcs_instance_guid
+  roles                       = ["Reader"]
+}
+
+resource "ibm_iam_authorization_policy" "secondary_kms_policy" {
+  count                       = var.skip_iam_authorization_policy ? 0 : 1
+  source_service_name         = "cloud-object-storage"
+  source_resource_instance_id = module.cos_instance.cos_instance_guid
+  target_service_name         = "hs-crypto"
+  target_resource_instance_id = var.secondary_existing_hpcs_instance_guid
+  roles                       = ["Reader"]
+}
+
+module "cos_primary_bucket" {
+  source                             = "../../"
+  resource_group_id                  = var.resource_group_id
+  region                             = var.primary_region
+  create_cos_instance                = false
+  existing_cos_instance_id           = module.cos_instance.cos_instance_id
+  create_cos_bucket                  = var.create_cos_bucket
+  bucket_name                        = "${var.bucket_name}-${var.primary_region}"
+  bucket_storage_class               = var.bucket_storage_class
+  retention_enabled                  = false
+  archive_days                       = null
+  expire_days                        = null
+  object_versioning_enabled          = "true"
+  existing_key_protect_instance_guid = var.primary_existing_hpcs_instance_guid
+  key_protect_key_crn                = var.primary_hpcs_key_crn
+  encryption_enabled                 = "true"
+  activity_tracker_crn               = var.activity_tracker_crn
+  sysdig_crn                         = var.sysdig_crn
+  bucket_cbr_rules                   = var.bucket_cbr_rules
+}
+
+module "cos_secondary_bucket" {
+  source                             = "../../"
+  resource_group_id                  = var.resource_group_id
+  region                             = var.secondary_region
+  create_cos_instance                = false
+  existing_cos_instance_id           = module.cos_instance.cos_instance_id
+  create_cos_bucket                  = var.create_cos_bucket
+  bucket_name                        = "${var.bucket_name}-${var.secondary_region}"
```

Comment:
Consumers are going to want the ability to name their buckets as they wish. I think this needs to move to a variable

Reply:
I still think we need to give more flexibility on naming here. Teams have request full control over naming. So can we have two required variables, `var.prinary_bucket_name` and `var.secondary_bucket_name`